music_assistant/server/providers/deezer/__init__.py

```diff
@@ -0,0 +1,355 @@
+        """Get all tracks in a playlist."""
+        playlist = await get_playlist(client=self.client, playlist_id=prov_playlist_id)
+        for track in playlist.tracks:
+            yield await parse_track(mass=self, track=track)
+
+    async def get_artist_albums(self, prov_artist_id: str) -> list[Album]:
+        """Get albums by an artist."""
+        artist = await get_artist(client=self.client, artist_id=int(prov_artist_id))
+        albums = []
+        for album in await get_albums_by_artist(artist=artist):
+            albums.append(await parse_album(mass=self, album=album))
+        return albums
+
+    async def get_artist_toptracks(self, prov_artist_id: str) -> list[Track]:
+        """Get top 25 tracks of an artist."""
+        artist = await get_artist(client=self.client, artist_id=int(prov_artist_id))
+        top_tracks = (await get_artist_top(artist=artist))[:25]
+        return [await parse_track(mass=self, track=track) for track in top_tracks]
+
+    async def library_add(self, prov_item_id: str, media_type: MediaType) -> bool:
+        """Add an item to the library."""
+        result = False
+        if media_type == MediaType.ARTIST:
+            result = await add_user_artists(
+                artist_id=int(prov_item_id),
+                client=self.client,
+            )
+        elif media_type == MediaType.ALBUM:
+            result = await add_user_albums(
+                album_id=int(prov_item_id),
+                client=self.client,
+            )
+        elif media_type == MediaType.TRACK:
+            result = await add_user_tracks(
+                track_id=int(prov_item_id),
+                client=self.client,
+            )
+        else:
+            raise NotImplementedError
+        return result
+
+    async def library_remove(self, prov_item_id: str, media_type: MediaType) -> bool:
+        """Remove an item to the library."""
+        result = False
+        if media_type == MediaType.ARTIST:
+            result = await remove_user_artists(
+                artist_id=int(prov_item_id),
+                client=self.client,
+            )
+        elif media_type == MediaType.ALBUM:
+            result = await remove_user_albums(
+                album_id=int(prov_item_id),
+                client=self.client,
+            )
+        elif media_type == MediaType.TRACK:
+            result = await remove_user_tracks(
+                track_id=int(prov_item_id),
+                client=self.client,
+            )
+        else:
+            raise NotImplementedError
+        return result
+
+    async def get_stream_details(self, item_id: str) -> StreamDetails | None:
+        """Return the content details for the given track when it will be streamed."""
+        track = await get_track(client=self.client, track_id=int(item_id))
+        return StreamDetails(
+            item_id=item_id,
+            provider=self.instance_id,
+            content_type=ContentType.MP3,
```

Comment:
this should be the real ContentType of the track

Reply:
Yeah

Reply:
Fixed in newest commit